### ISSUE TITLE
Feature Request: Support hyphenated tag names

### ISSUE THREAD

**Issue by cmaher** [wix/react-templates#9]
The custom-elements spec requires that custom elements contain a hyphen in the name. React natively supports this with `React.createElement("my-custom-element", null)`.  Currently, react-templates compiles `<my-custom-element>` as `React.createElement(my - custom - element, {})`, which is undesirable.

Additionally, it would be nice to support hyphenated identifiers for components imported via `rt-require`:  `<rt-require dependency='users/form' as='users-form'/>`.

This could be handled in the templating phase (pre esprima.parse), and two separate cases would need to be handled:

1. For hyphens in rt-require components: convert dash-case names to camelCase.
2. For custom elements, enclose the identifier in quotes.

I'd be willing to work on this, but I figure I'd check to see if there were interest in the idea before starting development.

**Comment by idok** [comment]
Regarding the custom elements, it was fixed in commit 5015222f4577a4898e3c353aa2d2dbade50979c3. It will be available in next release.
Regarding the second issue, we fill it will be confusing as the element and variable will not be the same.

**Comment by cmaher** [comment]
I don't think the variable naming scheme will be confusing. Angular already does this with their directives (matching `users-form` to `usersForm`), and only advanced users will have use of the generated code anyway.

so, for example

```
<rt-require dependency='users/form' as='users-form'/>
<users-form/>

//Generated code
var usersForm = require('users/form');
...
React.createElement(usersForm, {})
```
As long as a consistent, documented approach is taken, users should be fine with it.
